openedx/core/djangoapps/content/block_structure/config/__init__.py

```diff
@@ -10,12 +10,41 @@
 from .models import BlockStructureConfiguration
 
 # Switches
+# .. toggle_name: INVALIDATE_CACHE_ON_PUBLISH
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Invalidates block structure cache when changes to courses are published.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14571
 INVALIDATE_CACHE_ON_PUBLISH = WaffleSwitch(
     "block_structure.invalidate_cache_on_publish", __name__
 )
+
+# .. toggle_name: STORAGE_BACKING_FOR_CACHE
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Enables retrieving block structures from storage instead of regenerating
+#   block structures when they aren't available in cache.
+# .. toggle_warnings: Depends on the storage class, `STORAGE_CLASS`, and storage keyword arguments,
+#   `STORAGE_KWARGS`, specified under `BLOCK_STRUCTURES_SETTINGS`.
+# .. toggle_use_cases: opt_in
+# .. toggle_creation_date: 2017-02-23
+# .. toggle_tickets: https://github.com/edx/edx-platform/pull/14512
 STORAGE_BACKING_FOR_CACHE = WaffleSwitch(
     "block_structure.storage_backing_for_cache", __name__
 )
+
+# .. toggle_name: RAISE_ERROR_WHEN_NOT_FOUND
+# .. toggle_implementation: WaffleSwitch
+# .. toggle_default: False
+# .. toggle_description: Raises an error if the requested block structure does not exist in block
+#   structure store, or if it is outdated. Block structure store refers to both cache and storage,
+#   if enabled.
+# .. toggle_use_cases: temporary
```

Comment:
I know this sounds weird, since this has been there for 4 years, but there's a [comment](https://github.com/edx/edx-platform/pull/14512/files#r103130819) from @nasthagiri, author of the `RAISE_ERROR_WHEN_NOT_FOUND` change mentioning that the switch is temporary in #14512.

> **Note to reviewers**: We now plan to raise an error instead, if the block-structure is not available when requested. This is now an unexpected situation since the block-structure now has permanent storage. We honor the `RAISE_ERROR_WHEN_NOT_FOUND` waffle switch in the interim as we transition over to the new strategy.

Reply:
@nasthagiri: Is this a candidate for DEPR now? Do you know if the _transition_ referred to in the comment above is complete?
@nizarmah: I would add a `.. toggle_warning` with a note that this flag is deprecated and will be removed at some point.  For target date, I am fine with using the 3 month rule from https://edx.readthedocs.io/projects/edx-toggles/en/latest/how_to/documenting_new_feature_toggles.html from the creation date.  This will potentially just bring this to someone's attention as very ready to DEPR and remove.  We'll see if @nasthagiri feels otherwise.
Thank you.

Reply:
Thanks for the quick reply @robrap! I followed your suggestions with the commit [`b7fa3be`](https://github.com/edx/edx-platform/pull/26175/commits/b7fa3bec84db96278f8fb50d6926d12b8d648487).

I'll keep my eye out for a reply from @nasthagiri, and apply the necessary changes 👍🏼

Reply:
Great.
1. I think it is toggle_warning (singular).
2. Consider adding a link to the comment you posted from Nimisha for more details in the warning.
2. If we don’t hear back tomorrow, just mark this resolved.
Thanks.

Reply:
> I think it is toggle_warning (singular).

I folowed [`feature_toggle_annotations.yaml`](https://github.com/edx/code-annotations/blob/70d076aa843ce5ca688e08623e6d40bc6d4aa241/code_annotations/contrib/config/feature_toggle_annotations.yaml#L20).

> Consider adding a link to the comment you posted from Nimisha for more details in the warning.

Yeah, that's a great idea 👍🏼 

> If we don’t hear back tomorrow, just mark this resolved.

Alright then 😃

Reply:
Yikes. Thanks.
@regisb: For warnings, we are using singular for settings and plural for toggles. See https://github.com/edx/code-annotations/blob/70d076aa843ce5ca688e08623e6d40bc6d4aa241/code_annotations/contrib/config/setting_annotations.yaml
I’d like this to be consistent. I probably prefer singular, but we probably have many more toggles, so we should probably go for simplest. Do you mind updating? Do you think we should add for ticket (or tickets?) optionally in settings?

Reply:
@nizarmah Thank you for raising the question regarding the `RAISE_ERROR_WHEN_NOT_FOUND` toggle. You rightly discovered that it was created to be used only for a short-term rollout period, with the intent to remove it.

So yes, this toggle should be DEPR-ed and removed. Looking in our production environment, though, I am a bit confused why it is not enabled. Perhaps there was a problem when it was turned on and so we kept it off.

At this point, given the age of this block_structure code, we'll want to just remove the toggle and the code (since it's been disabled this entire time).

Reply:
>  I probably prefer singular, but we probably have many more toggles, so we should probably go for simplest. Do you mind updating?

Yes, I prefer singular, too. But I'm not so keen on suddenly renaming the field, which would cause many linting failures. The right deprecation process would be the same as for databases:

1. Add a `.. toggle_warning:` field which would be handled the same way as `.. toggle_warnings:`, including in the Sphinx extension.
2. Rename all `.. toggle_warnings:` to `.. toggle_warning:`.
3. Drop the `.. toggle_warnings:` field.

Reply:
@nasthagiri thanks for your reply! That sounds good 👍🏼 
I've went with @robrap's suggestion to indicate its deprecation in the warning annotation.

Let me know, please, if you'd like me to make any further changes though 😄